packages/aws-cdk-lib/core/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable brace-style */
+import { Construct, IConstruct } from 'constructs';
+import { IAspect, Aspects } from './aspect';
+import * as cdk from '../../../aws-cdk-lib';
+import * as lambda from '../../aws-lambda';
+import { ReceiptRuleSet } from '../../aws-ses/lib/receipt-rule-set';
+import { Provider } from '../../custom-resources/lib';
+
+/**
+ * Runtime aspec
+ */
+abstract class RuntimeAspectsBase implements IAspect {
+
+  /**
+   * The string key for the runtime
+   */
+  public readonly targetRuntime: string;
+
+  constructor(runtime: string) {
+    this.targetRuntime = runtime;
+  }
+
+  public visit(construct: IConstruct): void {
+
+    //To handle custom providers
+    if (construct instanceof cdk.CustomResourceProviderBase) {
```

Comment:
I don't think we need to check for instances of `CustomResourceProviderBase`, `Provider`, or `ReceptRule`. At their core these are all the same construct: `CfnResource` of type `AWS::Lambda::Function`. This should be all we need to look for.

Reply:
This is more to the point that we want to update it only 'CustomResources' and not for all lambda functions which i think we can cover with `CfnResource` of type `AWS::Lambda::Function` .